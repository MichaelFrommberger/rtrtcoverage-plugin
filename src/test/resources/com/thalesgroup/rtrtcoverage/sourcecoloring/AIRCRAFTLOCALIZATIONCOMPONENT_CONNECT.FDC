VE 3.0e CCOV #i5 7.0.0.1
FC "AIRCRAFTLOCALIZATIONCOMPONENT_CONNECT.C" "R:\\DGWP-SW-A350\\DGWP_A350_UT\\DGWP_ANF\\CLASS_AIRCRAFTLOCALIZATIONCOMPONENT\\SOURCES\\"
FD 1331629882
TC PR RE BS BT BI BL
BL 0 simple "AircraftLocalizationComponent_connect" /case 38 38
BL 1 simple "AircraftLocalizationComponent_connect" /case 46 46
BL 2 simple "AircraftLocalizationComponent_connect" /case 54 54
BL 3 simple "AircraftLocalizationComponent_connect" /case 62 62
BL 4 simple "AircraftLocalizationComponent_connect" /case 70 70
BL 5 simple "AircraftLocalizationComponent_connect" /case 78 78
BL 6 simple "AircraftLocalizationComponent_connect" /default 86 86
PR 0 return "AircraftLocalizationComponent_connect" 92 92
PR 1 proc "AircraftLocalizationComponent_connect" 31 93
BL 7 simple "AircraftLocalizationComponent_connect" / 31 93
DC 1362319514
@RIK
  ENCODING="System"
  TOKEN="CCOV"
  SRC="AIRCRAFTLOCALIZATIONCOMPONENT_CONNECT.C"
  DIR="R:\DGWP-SW-A350\DGWP_A350_UT\DGWP_ANF\CLASS_AIRCRAFTLOCALIZATIONCOMPONENT\SOURCES\"
  DATE=1331629882
  CHECKSUM=1362319514
  SUMS
   ("&Functions"=0,
    "-&Exits"=1,
    "Functions and exits"=0+1),
   ("&Calls"),
   ("Statement &blocks"=20,
    "-&Implicit blocks"=21,
    "Decisions"=20+21,
    "&Loops"=22),
   ("Basic &conditions",
    "&Modified conditions",
    "M&ultiple conditions")
  ERRORS
    32="a source object was not linked with the right runtime object (check instrumentation mode)",
    49="a composition of conditions evaluates in an unexpected result, contact your vendor",
    50="a composition of conditions reached with an unexpected input vector, contact your vendor"
NODE TYPE=FILE NAME="AIRCRAFTLOCALIZATIONCOMPONENT_CONNECT.C"@@COMMENT@/* *************************************************************************************** */@/COMMENT@
@COMMENT@/* (C) Copyright by THALES Avionics                                                        */@/COMMENT@
@COMMENT@/* All rights reserved                                                                     */@/COMMENT@
@COMMENT@/*                                                                                         */@/COMMENT@
@COMMENT@/* This program is the property of THALES Avionics, LE HAILLAN-BORDEAUX FRANCE, and can    */@/COMMENT@
@COMMENT@/* only be used and copied with the prior written authorisation of THALES Avionics.        */@/COMMENT@
@COMMENT@/*                                                                                         */@/COMMENT@
@COMMENT@/* Any whole or partial copy of this program in either its original form or in a modified  */@/COMMENT@
@COMMENT@/* form must mention this copyright and its owner.                                         */@/COMMENT@
@COMMENT@/* *************************************************************************************** */@/COMMENT@
@COMMENT@/* PROJECT: ANF_A350                                                                       */@/COMMENT@
@COMMENT@/* *************************************************************************************** */@/COMMENT@
@COMMENT@/* FILE NAME: AircraftLocalizationComponent_connect.c                                      */@/COMMENT@
@COMMENT@/* *************************************************************************************** */@/COMMENT@

@COMMENT@#include "AircraftLocalizationComponent.h"@/COMMENT@

@COMMENT@#define __CLASS__ 	"AircraftLocalizationComponent"@/COMMENT@
@COMMENT@#define __METHOD__ 	"connect"@/COMMENT@

@COMMENT@/* Start of user includes */@/COMMENT@
@COMMENT@/* End of user includes */@/COMMENT@


@COMMENT@/* *************************************************************************************** */@/COMMENT@
@COMMENT@/* METHOD DESCRIPTION: [public][instance method]                                           */@/COMMENT@
@COMMENT@/* -------------------                                                                     */@/COMMENT@
@COMMENT@/* Connect the given feature instance to the named receptacle                              */@/COMMENT@
@COMMENT@/* *************************************************************************************** */@/COMMENT@
void @NODE TYPE=FUNCTION NAME="AircraftLocalizationComponent_connect"@@BRANCH MARK=TP ID=1 SUM=0@@LINK NAME="AircraftLocalizationComponent_connect"@AircraftLocalizationComponent_connect@/BRANCH@(AircraftLocalizationComponent *pThis, const T_char *pName, T_PortInterface pInterface)
@BRANCH MARK=TB ID=7 SUM=20@{
    @COMMENT@/* Start of user code */@/COMMENT@
	T_uint32 lReceptacleIndex = pThis->_componentHelper.indexOfReceptacle(&(pThis->_componentHelper), pName);

	@COMMENT@/* Update interface instance corresponding to receptacle index */@/COMMENT@
	@DECISION SUM=20+21 ID=0|1|2|3|4|5|6@switch (lReceptacleIndex)
	{
		@BRANCH MARK=TB ID=0 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_DATABASE_SERVICES_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.databaseServices = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=1 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_ERROR_THROWER_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.errorThrower = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=2 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_FLIGHT_PLAN_MANAGEMENT_SERVICES_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.flightPlanManagementServices = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=3 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_MAP_OBJECT_SERVICES_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.mapObjectServices = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=4 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_DATA_STORAGE_SERVICES_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.dataStorageServices = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=5 SUM=20@@BRANCH NO@case E_AIRCRAFT_LOCALIZATION_TASKS_BENCHMARKING_RECEPTACLE_INDEX:@/BRANCH@
		{
			pThis->_receptacles.benchmarking = pInterface;
			@COMMENT@/* TODO STUB L1: to uncomment for L2 */@/COMMENT@
			@COMMENT@/* pThis->_componentHelper.newConnection(&(pThis->_componentHelper)); */@/COMMENT@
		}
		break;@SHORTCUT MARK=TB ID=0@

		@BRANCH MARK=TB ID=6 SUM=20@@BRANCH NO@default:@/BRANCH@
			; @COMMENT@/* Nothing to do */@/COMMENT@
		break;@SHORTCUT MARK=TB ID=0@
	}@/BRANCH@@/BRANCH@@/BRANCH@@/BRANCH@@/BRANCH@@/BRANCH@@/BRANCH@@/DECISION@

    @COMMENT@/* End of user code */@/COMMENT@
	@BRANCH MARK=TP ID=0 SUM=1@return;@SHORTCUT MARK=TB ID=7@@/BRANCH@
}@/BRANCH@@/NODE@
@/NODE /RIK@
