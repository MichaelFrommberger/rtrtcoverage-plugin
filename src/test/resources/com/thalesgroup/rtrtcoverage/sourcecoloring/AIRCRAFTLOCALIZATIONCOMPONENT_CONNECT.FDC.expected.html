<html><head></head><body><table class="rtrtcoverage-file">
<style media="screen" type="text/css">
.rtrtcoverage-complete {
  background-color: #ccffcc;}
.rtrtcoverage-partial {
  background-color: #ffffcc;}
.rtrtcoverage-none {
  background-color: #ffcccc;}
.rtrtcoverage-hint {
  border: 1px dashed grey;}
</style>
<tr><td><pre>1</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>2</pre></td><td><pre><span>/* (C) Copyright by THALES Avionics                                                        */</span></pre></td></tr>
<tr><td><pre>3</pre></td><td><pre><span>/* All rights reserved                                                                     */</span></pre></td></tr>
<tr><td><pre>4</pre></td><td><pre><span>/*                                                                                         */</span></pre></td></tr>
<tr><td><pre>5</pre></td><td><pre><span>/* This program is the property of THALES Avionics, LE HAILLAN-BORDEAUX FRANCE, and can    */</span></pre></td></tr>
<tr><td><pre>6</pre></td><td><pre><span>/* only be used and copied with the prior written authorisation of THALES Avionics.        */</span></pre></td></tr>
<tr><td><pre>7</pre></td><td><pre><span>/*                                                                                         */</span></pre></td></tr>
<tr><td><pre>8</pre></td><td><pre><span>/* Any whole or partial copy of this program in either its original form or in a modified  */</span></pre></td></tr>
<tr><td><pre>9</pre></td><td><pre><span>/* form must mention this copyright and its owner.                                         */</span></pre></td></tr>
<tr><td><pre>10</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>11</pre></td><td><pre><span>/* PROJECT: ANF_A350                                                                       */</span></pre></td></tr>
<tr><td><pre>12</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>13</pre></td><td><pre><span>/* FILE NAME: AircraftLocalizationComponent_connect.c                                      */</span></pre></td></tr>
<tr><td><pre>14</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>15</pre></td><td><pre></pre></td></tr>
<tr><td><pre>16</pre></td><td><pre><span>#include &quot;AircraftLocalizationComponent.h&quot;</span></pre></td></tr>
<tr><td><pre>17</pre></td><td><pre></pre></td></tr>
<tr><td><pre>18</pre></td><td><pre><span>#define __CLASS__ 	&quot;AircraftLocalizationComponent&quot;</span></pre></td></tr>
<tr><td><pre>19</pre></td><td><pre><span>#define __METHOD__ 	&quot;connect&quot;</span></pre></td></tr>
<tr><td><pre>20</pre></td><td><pre></pre></td></tr>
<tr><td><pre>21</pre></td><td><pre><span>/* Start of user includes */</span></pre></td></tr>
<tr><td><pre>22</pre></td><td><pre><span>/* End of user includes */</span></pre></td></tr>
<tr><td><pre>23</pre></td><td><pre></pre></td></tr>
<tr><td><pre>24</pre></td><td><pre></pre></td></tr>
<tr><td><pre>25</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>26</pre></td><td><pre><span>/* METHOD DESCRIPTION: [public][instance method]                                           */</span></pre></td></tr>
<tr><td><pre>27</pre></td><td><pre><span>/* -------------------                                                                     */</span></pre></td></tr>
<tr><td><pre>28</pre></td><td><pre><span>/* Connect the given feature instance to the named receptacle                              */</span></pre></td></tr>
<tr><td><pre>29</pre></td><td><pre><span>/* *************************************************************************************** */</span></pre></td></tr>
<tr><td><pre>30</pre></td><td><pre><span>void </span><span class="rtrtcoverage-complete">AircraftLocalizationComponent_connect</span><span>(AircraftLocalizationComponent *pThis, const T_char *pName, T_PortInterface pInterface)</span></pre></td></tr>
<tr><td><pre>31</pre></td><td><pre><span class="rtrtcoverage-complete">{</span></pre></td></tr>
<tr><td><pre>32</pre></td><td><pre><span class="rtrtcoverage-complete">    </span><span>/* Start of user code */</span></pre></td></tr>
<tr><td><pre>33</pre></td><td><pre><span class="rtrtcoverage-complete">	T_uint32 lReceptacleIndex = pThis-&gt;_componentHelper.indexOfReceptacle(&amp;(pThis-&gt;_componentHelper), pName);</span></pre></td></tr>
<tr><td><pre>34</pre></td><td><pre></pre></td></tr>
<tr><td><pre>35</pre></td><td><pre><span class="rtrtcoverage-complete">	</span><span>/* Update interface instance corresponding to receptacle index */</span></pre></td></tr>
<tr><td><pre>36</pre></td><td><pre><span class="rtrtcoverage-complete">	</span><span class="rtrtcoverage-complete">switch (lReceptacleIndex)</span></pre></td></tr>
<tr><td><pre>37</pre></td><td><pre><span class="rtrtcoverage-complete">	{</span></pre></td></tr>
<tr><td><pre>38</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_DATABASE_SERVICES_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>39</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>40</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.databaseServices = pInterface;</span></pre></td></tr>
<tr><td><pre>41</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>42</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>43</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>44</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>45</pre></td><td><pre></pre></td></tr>
<tr><td><pre>46</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_ERROR_THROWER_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>47</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>48</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.errorThrower = pInterface;</span></pre></td></tr>
<tr><td><pre>49</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>50</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>51</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>52</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>53</pre></td><td><pre></pre></td></tr>
<tr><td><pre>54</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_FLIGHT_PLAN_MANAGEMENT_SERVICES_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>55</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>56</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.flightPlanManagementServices = pInterface;</span></pre></td></tr>
<tr><td><pre>57</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>58</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>59</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>60</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>61</pre></td><td><pre></pre></td></tr>
<tr><td><pre>62</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_MAP_OBJECT_SERVICES_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>63</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>64</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.mapObjectServices = pInterface;</span></pre></td></tr>
<tr><td><pre>65</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>66</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>67</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>68</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>69</pre></td><td><pre></pre></td></tr>
<tr><td><pre>70</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_DATA_STORAGE_SERVICES_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>71</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>72</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.dataStorageServices = pInterface;</span></pre></td></tr>
<tr><td><pre>73</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>74</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>75</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>76</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>77</pre></td><td><pre></pre></td></tr>
<tr><td><pre>78</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">case E_AIRCRAFT_LOCALIZATION_TASKS_BENCHMARKING_RECEPTACLE_INDEX:</span></pre></td></tr>
<tr><td><pre>79</pre></td><td><pre><span class="rtrtcoverage-complete">		{</span></pre></td></tr>
<tr><td><pre>80</pre></td><td><pre><span class="rtrtcoverage-complete">			pThis-&gt;_receptacles.benchmarking = pInterface;</span></pre></td></tr>
<tr><td><pre>81</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* TODO STUB L1: to uncomment for L2 */</span></pre></td></tr>
<tr><td><pre>82</pre></td><td><pre><span class="rtrtcoverage-complete">			</span><span>/* pThis-&gt;_componentHelper.newConnection(&amp;(pThis-&gt;_componentHelper)); */</span></pre></td></tr>
<tr><td><pre>83</pre></td><td><pre><span class="rtrtcoverage-complete">		}</span></pre></td></tr>
<tr><td><pre>84</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>85</pre></td><td><pre></pre></td></tr>
<tr><td><pre>86</pre></td><td><pre><span class="rtrtcoverage-complete">		</span><span class="rtrtcoverage-complete">default:</span></pre></td></tr>
<tr><td><pre>87</pre></td><td><pre><span class="rtrtcoverage-complete">			; </span><span>/* Nothing to do */</span></pre></td></tr>
<tr><td><pre>88</pre></td><td><pre><span class="rtrtcoverage-complete">		break;</span></pre></td></tr>
<tr><td><pre>89</pre></td><td><pre><span class="rtrtcoverage-complete">	}</span></pre></td></tr>
<tr><td><pre>90</pre></td><td><pre></pre></td></tr>
<tr><td><pre>91</pre></td><td><pre><span class="rtrtcoverage-complete">    </span><span>/* End of user code */</span></pre></td></tr>
<tr><td><pre>92</pre></td><td><pre><span class="rtrtcoverage-complete">	</span><span class="rtrtcoverage-complete">return;</span></pre></td></tr>
<tr><td><pre>93</pre></td><td><pre><span class="rtrtcoverage-complete">}</span></pre></td></tr>
<tr><td><pre>94</pre></td><td><pre></pre></td>
</tr>
</table></body></html>
